pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -362,6 +389,7 @@ void messageReceived(Message message) {
             } else {
                 // Enqueue the message so that it can be retrieved when application calls receive()
                 // Waits for the queue to have space for the message
+                lock.readLock().lock();
```

Comment:
We should have already the `readLock` taken here (from line 379). This can lead to deadlock with the write lock.

Reply:
Yes, this should not be here.